packages/flutter_tools/lib/src/base/flags.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+
+import 'context.dart';
+
+Flags get flags => context?.getVariable(Flags) ?? const _EmptyFlags();
+
+class Flags {
+  Flags(this._globalResults) {
+    assert(_globalResults != null);
+  }
+
+  final ArgResults _globalResults;
+
+  dynamic operator [](String key) {
+    final ArgResults commandResults = _globalResults.command;
+    if (commandResults?.options?.contains(key) ?? false) {
```

Comment:
nit: `?? false` does work, but I think it's less readable than:

```
final options = commandResults?.options;
if (options != null && options.contains(key)) {
  ...
```

Reply:
+1 in both agreement and nitpickiness. ;-)